### ISSUE TITLE
Compile Issues On "master" and "stable"

### ISSUE THREAD

**Issue by ahrendsen** [Liam-Deacon/CLEED#1]
Hello,

I was interested in using this program for my research and was attempting to compile it. The instructions for compiling could use some updating, as there were a number of package prerequisites I didn't have on my system, and cmake no longer accepts -i as an option, which led to a whole lot of other weirdness. 

Even after getting those difficulties sorted out, it appeared that the package was not in a state so that it could be compiled. The first compile complained that lattice.c didn't have a main function or defined variables. The second compile revealed that a variable was likely renamed from "eng" to "energy" in one file, but not another. The current hurdle is that eng is being referenced as if it were a struct, but the compiler doesn't think that it should be. At this point, I began to question whether it was worth chasing down all of these threads.

I would really appreciate if you could provide a version of the program which compiles for you, just so I can know that the issues I'm experiencing are because of something wrong with my installation, and not because the project wasn't in a working state to begin with.

I also tried installing via pip, but it said "No matching distribution found for cleed"

I can provide more details of the issues I experienced, if that would be helpful for you.

Thanks in advance for your help,
Karl

**Comment by ahrendsen** [comment]
I've been slowly working through the errors, but I've hit one that I'm not sure how to go about solving involving the libmkiv.so library. See the full details in the commit message at https://github.com/ahrendsen/CLEED/commit/6f4476550e8f0b2c946523e9f573a8fea997e1c2

**Comment by ahrendsen** [comment]
Attempting to use the non-gui cmake, I run into issues that I can't resolve involving the previous place it was compiled.

```
-- Detecting Fortran/C Interface
CMake Error: The current CMakeCache.txt directory /home/karl/gitSpace/CLEED3/CLEED/CMakeFiles/FortranCInterface/CMakeCache.txt is different than the directory c:/Users/kss07698/Dropbox/Windows Tweaks/CLEED_tools_win32/cleed/src/CLEED/CMakeFiles/FortranCInterface where CMakeCache.txt was created. This may result in binaries being created in the wrong place. If you are not sure, reedit the CMakeCache.txt
CMake Error: The source "/usr/share/cmake-3.10/Modules/FortranCInterface/CMakeLists.txt" does not match the source "C:/Users/kss07698/Dropbox/Windows Tweaks/CLEED_tools_win32/cleed/src/CLEED/CMakeFiles/FortranCInterfaceMinGW/CMakeLists.txt" used to generate cache.  Re-run cmake with a different source directory.
CMake Error at /usr/share/cmake-3.10/Modules/FortranCInterface/Detect.cmake:31 (try_compile):
  Failed to configure test project build system.
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FortranCInterface.cmake:224 (include)
  CMakeLists.txt:210 (INCLUDE)


-- Configuring incomplete, errors occurred!
See also "/home/karl/gitSpace/CLEED3/CLEED/CMakeFiles/CMakeOutput.log".
See also "/home/karl/gitSpace/CLEED3/CLEED/CMakeFiles/CMakeError.log".
```
I'm sure there's an easy fix to this, but the output doesn't seem to be helpful.

**Comment by ahrendsen** [comment]
I don't know what I did to fix it, but the libmkv.so issue is no longer present. I know this is kind of haphazard right now, but it's mainly bookkeeping for myself and I'll link my fork once I have identified all of the changes that need to be made.

Now I'm encountering an issue with patt.c, it looks like something I can sort out. Here's a snippet of the errors:
```
[ 64%] Built target mkiv
[ 64%] Building C object src/patt/CMakeFiles/patt.dir/patt.c.o
/home/karl/gitSpace/CLEED3/CLEED/src/patt/patt.c: In function ‘main’:
/home/karl/gitSpace/CLEED3/CLEED/src/patt/patt.c:116:18: error: ‘PATT_COLOR’ undeclared (first use in this function); did you mean ‘PATT_GOLD’?
 int color_flag = PATT_COLOR;        /* PATT_COLOR for rgb (default) */
                  ^~~~~~~~~~
                  PATT_GOLD
/home/karl/gitSpace/CLEED3/CLEED/src/patt/patt.c:116:18: note: each undeclared identifier is reported only once for each function it appears in
/home/karl/gitSpace/CLEED3/CLEED/src/patt/patt.c:124:2: error: ‘out_stream’ undeclared (first use in this function)
  out_stream = stdout;
  ^~~~~~~~~~
/home/karl/gitSpace/CLEED3/CLEED/src/patt/patt.c:125:2: error: ‘in_stream’ undeclared (first use in this function); did you mean ‘out_stream’?
  in_stream = stdin;
  ^~~~~~~~~
  out_stream
/home/karl/gitSpace/CLEED3/CLEED/src/patt/patt.c:126:2: error: ‘in_streams’ undeclared (first use in this function); did you mean ‘in_stream’?
  in_streams[0] = stdin;
  ^~~~~~~~~~
  in_stream
```

**Comment by Liam-Deacon** [comment]
Closing as stale/resolved:  now configures/builds on modern toolchains and CI is green on linux/macos/windows. If you can still reproduce a build failure, please open a new issue with current logs (-- Java version check failed: /usr/bin/java -version returned an error: "The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java."
-- Could NOT find Java (missing: Java_JAVA_EXECUTABLE Java_JAVAH_EXECUTABLE) 
-- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES) 
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND) 
-- 
-- -------------------------------------------------------------------------------
-- BUILD_SHARED_LIBS = ON
-- CMAKE_INSTALL_PREFIX = /Users/liam/repos/cleed
-- CMAKE_INCLUDE_PATH = 
-- CMAKE_LIBRARY_PATH = 
-- CMAKE_MODULE_PATH = /Users/liam/repos/cleed/CMakeModules
-- CMAKE_BUILD_TYPE = Release
-- CMAKE_SYSTEM = Darwin-25.1.0
-- CMAKE_SYSTEM_PROCESSOR = arm64
-- CMAKE_C_COMPILER = /usr/bin/c++
-- CMAKE_CXX_COMPILER = /usr/bin/c++
-- CMAKE_Fortran_COMPILER = 
-- Java_JAVAC_EXECUTABLE = /usr/bin/javac
-- CMAKE_AR =  /usr/bin/ar
-- CMAKE_RANLIB =  /usr/bin/ranlib
-- CMAKE_C_FLAGS = 
-- CMAKE_CXX_FLAGS = 
-- CMAKE_C_FLAGS_RELEASE = 
-- CMAKE_CXX_FLAGS_RELEASE = 
-- CMAKE_EXE_LINKER_FLAGS =  -lm
-- CMAKE_MODULE_LINKER_FLAGS =  -lm
-- CMAKE_SHARED_LINKER_FLAGS =  -lm
-- CMAKE_C_COMPILER_VERSION = 17.0.0.17000404
-- CMAKE_CXX_COMPILER_VERSION = 17.0.0.17000404
-- COMPILE_DEFINITIONS = _USE_OPENCL;_USE_MATH_DEFINES
-- INSTALL_DOC = ON
-- WITH_OPENCL = ON
-- WITH_PHASESHIFTS = ON
-- WITH_QT = OFF
-- USE_OPENMP = OFF
-- PYTHON_INCLUDE_DIRS = /Library/Frameworks/Python.framework/Versions/3.11/include/python3.11
-- PYTHON_INCLUDE_PATH = /Library/Frameworks/Python.framework/Versions/3.11/include/python3.11
-- PYTHON_LIBRARIES = /Library/Frameworks/Python.framework/Versions/3.11/lib/libpython3.11.dylib
-- PYTHONLIBS_VERSION_STRING = 3.11.9
-- PYTHON_PIP_EXECUTABLE = /Users/liam/.pyenv/shims/pip
-- F2PY_EXECUTABLE = /opt/homebrew/bin/f2py
-- OPENCL_LIB_SEARCH_PATH = /lib/x86_64;/lib/x64;/OpenCL/common/lib/x64;/lib/x86_64
-- OPENCL_INCLUDE_DIR = /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenCL.framework
-- OPENCL_LIBRARY = /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenCL.framework
-- 
-- Change a value with: cmake -D<Variable>=<Value>
-- -------------------------------------------------------------------------------
-- 
-- Configuring done (1.1s)
-- Generating done (0.2s)
-- Build files have been written to: /Users/liam/repos/cleed/build output, compiler/CMake versions, OS) and we’ll track it there.

**Comment by Liam-Deacon** [comment]
Follow-up: my previous close comment accidentally included local `cmake` output due to shell backtick expansion. Please ignore that pasted configure output.

Rationale for closing: current `master` configures/builds on modern toolchains and CI is green on linux/macos/windows. If you can still reproduce a build failure, please open a new issue with current logs (full `cmake -S . -B build -DCMAKE_BUILD_TYPE=Release` output), compiler/CMake versions, and OS.
